src/Program.ts

```diff
@@ -69,6 +70,10 @@ export class Program {
         this.logger = logger || new Logger(options.logLevel as LogLevel);
         this.plugins = plugins || new PluginInterface([], undefined);
 
+        //inject the bsc plugin as the first plugin in the stack.
+        //eslint-disable-next-line @typescript-eslint/dot-notation
+        this.plugins['plugins'].unshift(new BscPlugin());

```

Comment:
Shouldn't we have some "add first" method in `PluginInterface` instead?

Reply:
I want the BscPlugin to ALWAYS be first in the chain, an `addFirst` method would make it easier for external users to mess with the order of things. I don't see a reason for an `addFirst` method _except_ for what we are doing here...and since I'm using the index access pattern we will get tsc errors if the internals of `PluginInterface` ever change.

Reply:
A user plugin could use addFirst, and the compiler is always the last to add plugins.

Reply:
Fair enough. If a dev starts messing with the PluginInterface, it's their own fault.

Reply:
Done.